crates/bevy_asset/src/asset_server.rs

```diff
@@ -781,8 +781,11 @@ mod test {
             asset_server.get_handle_untyped(id)
         }
 
-        fn get_asset(id: impl Into<HandleId>, world: &World) -> Option<&PngAsset> {
-            world.resource::<Assets<PngAsset>>().get(id.into())
+        fn get_asset<'world>(
+            id: &Handle<PngAsset>,
+            world: &'world World,
+        ) -> Option<&'world PngAsset> {
```

Comment:
Note to reviewers: this is part of a test suite. Hence the overly broad name for a specific asset type.